x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/MappingsMerger.java

```diff
@@ -59,19 +83,11 @@ public static void mergeMappings(Client client, Map<String, String> headers, Dat
             Iterator<ObjectObjectCursor<String, MappingMetadata>> typeIterator = indexMappings.value.iterator();
             while (typeIterator.hasNext()) {
                 ObjectObjectCursor<String, MappingMetadata> typeMapping = typeIterator.next();
-                if (type == null) {
```

Comment:
This was a tricky backport because of the difference between `master` and `7.x` due to type removal.

Here we used to check the type names are all the same. I decided to simplify this bit by removing this validation. It is not really a meaningful validation. The reason is that in `7.x` each index may only contain a single type. While it is possible for 6.x indices to have differently named types, as long as their mappings do not conflict we should be fine.

Reply:
Yes, sounds fine to me too